### PR TITLE
Changes for Catalog module

### PR DESCRIPTION
**Pull request by swati21**
- Addition of cascade delete
- Testcase for relevant functionalities
- Changes incorporated as suggested in previous PR

### CODE REVIEW

File: src/catalog/services/df_service.py [hunk 1]
```diff
@@ -87,3 +87,19 @@ def dataset_object_by_name(self, database_name, dataset_name,
         """
         return self.model.query.filter(
             self.model._name == dataset_name).one()
+
```
Comment (gaurav274) [inline src/catalog/services/df_service.py:90]: Add test case for this implementation

File: src/catalog/models/base_model.py [hunk 2]
```diff
@@ -74,9 +74,12 @@ def update(self, **kwargs):
 
     def delete(self):
         """Delete and commit"""
-
-        db_session.delete(self)
-        self._commit()
+        try:
```
Comment (gaurav274) [inline src/catalog/models/base_model.py:77]: Add test case for this function. I think we don't have any unit testing for this file. Can you please create on and add test cases?

File: src/catalog/services/df_service.py (hunk 1 above)
Comment (swati21) [reply src/catalog/services/df_service.py:90]: def (test_dataset_object_by_name_queries_with_name_returns_model_object) this testcasse is already present in the test_dataset_service.py

File: src/catalog/models/base_model.py (hunk 2 above)
Comment (swati21) [reply src/catalog/models/base_model.py:77]: Yes we will need to create testcases for Base Model. Will add that

**Pull request closed without merging**